sisl/viz/plots/bands.py

```diff
@@ -347,17 +356,15 @@ def _read_from_H(self, band_structure, eigenstate_map):
 
         # Also, in this wrapper we will get the spin moments in case it is a non_colinear
         # or spin-orbit calculation
-        if self.spin.is_noncolinear or self.spin.is_spinorbit:
-            self.spin_moments = []
-        elif hasattr(self, "spin_moments"):
-            del self.spin_moments
+        calc_spin_moments = not self.spin.is_diagonal
```

Comment:
couldn't you just append to `extra_vars` to clean out this?

Reply:
Yes!